src/swap/saga.ts

```diff
@@ -0,0 +1,114 @@
+} from 'src/swap/slice'
+import { Field, SwapInfo, SwapTransaction } from 'src/swap/types'
+import { sendTransaction } from 'src/transactions/send'
+import { newTransactionContext } from 'src/transactions/types'
+import { fetchWithTimeout } from 'src/utils/fetchWithTimeout'
+import Logger from 'src/utils/Logger'
+import { getContractKit } from 'src/web3/contracts'
+import networkConfig from 'src/web3/networkConfig'
+import { getConnectedUnlockedAccount } from 'src/web3/saga'
+import { applyChainIdWorkaround, buildTxo } from 'src/web3/utils'
+
+const TAG = 'swap/saga'
+
+function getPercentageDifference(price1: number, price2: number) {
+  return (Math.abs(price1 - price2) / ((price1 + price2) / 2)) * 100
+}
+
+export function* swapSubmitSaga(action: PayloadAction<SwapInfo>) {
+  try {
+    // Navigate to swap pending screen
+    yield call(navigate, Screens.SwapPending)
+
+    // Check that our guaranteedPrice is within 2%, maxSwapSlippagePercentage, of of the price
+    const maxSlippagePercent: number = yield select(maxSwapSlippagePercentageSelector)
+    const priceDiff: number = yield call(
+      getPercentageDifference,
+      +action.payload.unvalidatedSwapTransaction.price,
+      +action.payload.unvalidatedSwapTransaction.guaranteedPrice
+    )
+    if (priceDiff >= maxSlippagePercent) {
+      yield put(swapPriceChange())
+      return
+    }
+
+    // Set contract kit, wallet address and normalizer
+    const kit: ContractKit = yield call(getContractKit)
+    const walletAddress: string = yield call(getConnectedUnlockedAccount)
+    const normalizer = new TxParamsNormalizer(kit.connection)
+
+    // Approve transaction
+    yield put(swapApprove())
+    Logger.debug(TAG, `Starting to swap approval for address: ${walletAddress}`)
+    const rawApproveTx = { ...action.payload.approveTransaction }
+    applyChainIdWorkaround(rawApproveTx, yield call([kit.connection, 'chainId']))
+    const approveTx: CeloTx = yield call(normalizer.populate.bind(normalizer), rawApproveTx)
+    const approveTxo = buildTxo(kit, approveTx)
+    yield call(
+      sendTransaction,
+      approveTxo,
+      walletAddress,
+      newTransactionContext(TAG, 'Swap/Approve')
+    )
+
+    // Query the execute swap endpoint
+    const amountType =
+      action.payload.userInput.updatedField === Field.TO ? 'buyAmount' : 'sellAmount'
+    const amount = action.payload.unvalidatedSwapTransaction[amountType]
+    const params = {
+      buyToken: action.payload.unvalidatedSwapTransaction.buyTokenAddress,
+      sellToken: action.payload.unvalidatedSwapTransaction.sellTokenAddress,
+      [amountType]: amount,
+      userAddress: walletAddress,
+    }
+    const queryParams = new URLSearchParams({ ...params }).toString()
+    const requestUrl = `${networkConfig.executeSwapUrl}?${queryParams}`
+    const response: Response = yield call(fetchWithTimeout, requestUrl)
+    if (!response.ok) {
+      Logger.error(TAG, `Swap failed with status: ${response.status}`)
```

Comment:
let's not alert sentry about failed swaps...

Reply:
also, instead of handling the error here, why not just throw and let the catch on line 95 handle it?

Reply:
Fixed in 8fb182bf7fb0da7823c739f28480e7f2cc7947dd!